scripts/reliability_metrics.py

```diff
@@ -270,45 +72,72 @@ def init_states():
         StateFactory.register_state('FAULT', FaultState)
 
 
+class Utils:
+    args: argparse.Namespace = None
+    handler: logging.Handler = None
+    loggers: dict = {}
+    logger_adapters: dict = {}
 
-def get_arguments() -> argparse.Namespace:
-    parser: argparse.ArgumentParser = argparse.ArgumentParser(description='find services errors in syslog')
-    parser.add_argument('state', choices=['MASTER', 'BACKUP', 'FAULT'], help='state of the machine')
-    return parser.parse_args()
+    @staticmethod
+    def get_arguments() -> argparse.Namespace:
+        if Utils.args is not None:
+            return Utils.args
+        parser: argparse.ArgumentParser = argparse.ArgumentParser(description='find services errors in syslog')
+        parser.add_argument('state', choices=['MASTER', 'BACKUP', 'FAULT'], help='state of the machine')
+        Utils.args = parser.parse_args()
+        return Utils.args
 
+    @staticmethod
+    def get_file_handler() -> logging.Handler:
+        if Utils.handler is not None:
+            return Utils.handler
 
-def get_logger(name: str) -> logging.Logger:
-    log_file_name: str = os.path.basename(__file__).split('.')[0] + '.log'
-    formatter: logging.Formatter = logging.Formatter('[%(asctime)s] [%(levelname)s] [%(name)s] [%(funcName)s:%(lineno)d] %(message)s')
-    handler: logging.Handler = logging.handlers.RotatingFileHandler(log_file_name, maxBytes=0x10000,
-                                  backupCount=10)
-    handler.setFormatter(formatter)
-    logger = logging.getLogger(name)
-    logger.addHandler(handler)
+        file_name: str = f'{os.path.sep}'.join(('', 'tmp', os.path.splitext(os.path.basename(__file__))[0] + '.log'))
+        formatter: logging.Formatter = logging.Formatter('[%(asctime)s] [%(levelname)s] [%(name)s] [%(thread)d] [%(funcName)s:%(lineno)d] %(message)s')
+        Utils.handler: logging.Handler = logging.handlers.RotatingFileHandler(file_name, maxBytes=0x10000,
+                                      backupCount=10)
+        Utils.handler.setFormatter(formatter)
+        return Utils.handler
 
-    return logger
+    @staticmethod
+    def get_logger(name: str) -> logging.Logger:
+        if name in Utils.loggers:
+            return Utils.loggers[name]
+        logger: logging.Logger = logging.getLogger(name)
+        handler: logging.Handler = Utils.get_file_handler()
+        logger.setLevel(logging.INFO)
```

Comment:
is level always-always info?

Reply:
need to know basis .. nothing more, nothing less

Reply:
I agree with @sagi-ottopia